src/sentry/api/serializers/models/userreport.py

```diff
@@ -1,25 +1,18 @@
+from sentry import eventstore
 from sentry.api.serializers import Serializer, register, serialize
-from sentry.models.eventuser import EventUser
 from sentry.models.group import Group
 from sentry.models.userreport import UserReport
+from sentry.utils.eventuser import EventUser
 
 
 @register(UserReport)
 class UserReportSerializer(Serializer):
     def get_attrs(self, item_list, user, **kwargs):
-        event_user_ids = {i.event_user_id for i in item_list if i.event_user_id}
-
-        # Avoid querying if there aren't any to actually query, it's possible
-        # for event_user_id to be None.
-        if event_user_ids:
-            queryset = list(EventUser.objects.filter(id__in=event_user_ids))
-            event_users = {e.id: d for e, d in zip(queryset, serialize(queryset, user))}
-        else:
-            event_users = {}
-
         attrs = {}
+
         for item in item_list:
-            attrs[item] = {"event_user": event_users.get(item.event_user_id)}
+            event = eventstore.backend.get_event_by_id(item.project_id, item.event_id)
```

Comment:
This is an n+1, you'll need to figure out how to batch it. There should be bulk event fetching methods around